Documentation of end-entity sign/seal certificate profiles (#88)
APTITUDE-Consortium/wp2-trust-specifications: docs/topics/entity-end-certificate-profiles.md

| `keyUsage` | [ETSI TS 119 412-6, clause 4.4.1] | REQUIRED | C | *BIT STRING* | It SHOULD contain one (and only one) of the key-usage settings Type A, Type B, Type C, or Type F, as defined in ETSI EN 319 412-2. |
| `subjectKeyIdentifier` | [ETSI TS 119 412-6, clause 4.4.2] | REQUIRED | NC | *BIT STRING* | For end entity certificates, the subject key identifier extension provides a means of identifying certificates that contain the particular public key used in an application. The subject key identifier SHOULD be derived from the public key using the methods defined in RFC 5280, clause 4.2.1.2. |
| `authorityInfoAccess` | [ETSI TS 119 412-6, clause 4.4.3] | REQUIRED (C) | NC | *SEQUENCE* | Description is the same as in the General Content above. **Applicable condition:** Mandatory for non-self-signed certificates. |
| `qcStatements` (id-etsi-qct-pid) | [ETSI TS 119 412-6, clause 4.5] | REQUIRED | NC | *SEQUENCE* | `QCStatement` with the OID `0.4.0.194126.1.1` as defined in [ETSI TS 119 412-6, Annex A] |

#### PID Provider Sign/Seal Certificate Example

The following is an example of a PID Provider's non-self-signed end-certificate for legal persons.

```text

> AccessCertificate cert = {

Reviewer: Do we need to define the `AccessCertificate` type/structure? Or say that all sub-structures are standard certificate elements but the outer structure is just a placeholder? This type also appears in #78 and in one line in branch `main`, so this may not be specific to this PR.

Author: I have borrowed this structure from our documentation https://aptitude-consortium.github.io/wp2-trust-specifications/v0.1/deliverable-2.1-trust/#examples to be consistent. But I am open for any suggestion. For sure it should be changed everywhere where it is used.

Reviewer 2: We used this notation as it was similar to the one used in many DER-encoding certificate parsers like https://lapo.it/asn1js/.

We remain open for suggestions.

            accessLocation: URI("https://ca.example.test/caIssuers/issuing-ca.cer")
          },

          AccessDescription {
            accessMethod: "1.3.6.1.5.5.7.48.1",            // id-ad-ocsp
            accessLocation: URI("https://ocsp.example.test")
          }
        ]
      },

      Extension {
        oid: "2.5.29.32",            // certificatePolicies
        critical: false,
        value: CertificatePolicies [
          PolicyInformation {
            policyIdentifier: "0.4.0.194112.1.3",          // qcp-legal-qcsd
            policyQualifiers: [
              CPSuri("https://rpca.example.test/cps")
            ]
          }
        ]
      },

      Extension {
        oid: "1.3.6.1.5.5.7.0.35",   // qcStatements-2 container
        critical: false,
        value: QCStatements [
          QCStatement {
            statementId: "0.4.0.194126.1.3",   // id-etsi-qcs-QcPSB
            statementInfo: {
              countryOfLegislation: "CZ",
              authSourceIdentification: "https://authsource.gov.cz/cz/registry/rob",
              legislationIdentification: "https://legislation.gov.cz/eli/cz/sb/2000/365"

>             }

Author: ```suggestion
            }
          }
```

Author: @andmoro If I am not mistaken, we are now missing one bracket to close QC Statement, am I right?

| `qcStatements` (esi4-qcStatement-6) | [RFC 3739, clause 3.2.6] &<br>[ETSI EN 319 412-5, clause 4.2.3] | REQUIRED (C) | NC | *SEQUENCE* | `QCStatement` with the OID `0.4.0.1862.1.6`.<br><br>**Applicable condition:** Mandatory for qualified certificates issued to legal persons for the purpose of electronic seal ([ETSI EN 319 412-5, clause 5]). MAY be present for certificates issued to natural persons for the purpose of electronic signatures.<br><br>Declares that a certificate is issued for one and only one of the purposes: electronic signature, electronic seal, or website authentication. |

!!! note

    In the APTITUDE profiles, Sign/Seal Certificates SHALL be long lived certificates. Thus the `noRevAvail` and `ext-etsi-valassured-ST-certs` SHALL NOT be used. 

> ##### PID Provider Sing/Seal Certificate Content

Reviewer: ```suggestion
##### PID Provider Sign/Seal Certificate Content
```

This section describes the purpose, format and content of End Entity Sign/Seal Certificates in the European Digital Identity Wallet (EUDIW) ecosystem that are used for signing and sealing purposes.

#### References

To guarantee the interoperability across all entities of the EUDIW ecosystem, End Entity Sign/Seal Certificates should adhere to common requirements, with respect to their content and format. The technical specifications describing such content are distributed between multiple documents and for a purpose of proper referencing are listed below:

- **RFC 3986**
- **RFC 5280**
- **ETSI TS 119 612**
- **ETSI EN 319 411-1**
- **ETSI EN 319 411-2** (applicable if the certificate is qualified)
- **ETSI EN 319 412-1**
- **ETSI EN 319 412-2** (applicable if the certificate is issued to natural persons)
- **ETSI EN 319 412-3** (applicable if the certificate is issued to legal persons)
- **ETSI EN 319 412-5** (applicable if the certificate is qualified)

> - **ETSI TS 119 412-6** (applicable to the )

Author: @andmoro I think something is missing here in the brackets, right?

> - **ETSI TS 119 412-6** (applicable to the )

Reviewer 2: ```suggestion
- **ETSI TS 119 412-6** (profiles the Sign/Seal certificate for various entities in the EUDIW ecosystem)
```

        }
      },

      Extension {
        oid: "2.5.29.14",    // subject key identifier
        critical: false,
        value: SubjectKeyIdentifier [
          keyIdentifier: "SHA-1(SUBJECT_PUBLIC_KEY_VALUE)"
        ]
      },

      Extension {
        oid: "1.3.6.1.5.5.7.1.1",    // authority information access
        critical: false,
        value: AuthorityInfoAccess [
          AccessDescription {
            accessMethod: "1.3.6.1.5.5.7.48.2",            // id-ad-caIssuers
            accessLocation: URI("https://ca.example.test/caIssuers/issuing-ca.cer")
          },

          AccessDescription {
            accessMethod: "1.3.6.1.5.5.7.48.1",            // id-ad-ocsp
            accessLocation: URI("https://ocsp.example.test")
          }
        ]
      },

      Extension {
        oid: "2.5.29.32",            // certificatePolicies
        critical: false,
        value: CertificatePolicies [
          PolicyInformation {
            policyIdentifier: "0.4.0.194112.1.3",          // qcp-legal-qcsd
            policyQualifiers: [
              CPSuri("https://rpca.example.test/cps")
            ]
          }
        ]
      },

      Extension {

>         oid: "1.3.6.1.5.5.7.0.35",   // qcStatements-2 container

Reviewer 2: ```suggestion
        oid: "1.3.6.1.5.5.7.1.3",   // id-pe-qcStatements container
```

        }
      },

      Extension {
        oid: "2.5.29.14",    // subject key identifier
        critical: false,
        value: SubjectKeyIdentifier [
          keyIdentifier: "SHA-1(SUBJECT_PUBLIC_KEY_VALUE)"
        ]
      },

      Extension {
        oid: "1.3.6.1.5.5.7.1.1",    // authority information access
        critical: false,
        value: AuthorityInfoAccess [
          AccessDescription {
            accessMethod: "1.3.6.1.5.5.7.48.2",            // id-ad-caIssuers
            accessLocation: URI("https://ca.example.test/caIssuers/issuing-ca.cer")
          },

          AccessDescription {
            accessMethod: "1.3.6.1.5.5.7.48.1",            // id-ad-ocsp
            accessLocation: URI("https://ocsp.example.test")
          }
        ]
      },

      Extension {
        oid: "2.5.29.32",            // certificatePolicies
        critical: false,
        value: CertificatePolicies [
          PolicyInformation {
            policyIdentifier: "0.4.0.194112.1.3",          // qcp-legal-qcsd
            policyQualifiers: [
              CPSuri("https://rpca.example.test/cps")
            ]
          }
        ]
      },

      Extension {

>         oid: "1.3.6.1.5.5.7.0.35",   // qcStatements-2 container

Reviewer 2: ```suggestion
        oid: "1.3.6.1.5.5.7.1.3",   // id-pe-qcStatements container
```

Author: @andmoro Can you help me to understand where is this id-pe-qcStatements oid comming from? I may understood it wrongly, but in the ETSI TS 119 412-6 V1.1.1 (2025-09), Annex A, there is this container defined:
```
QC-STATEMENT, qcStatement-2
FROM PKIXqualified97 {iso(1) identified-organization(3) dod(6)
internet(1) security(5) mechanisms(5) pkix(7) id-mod(0)
id-mod-qualified-cert-97(35)};
```
From that, I have understood that OID for the container is 1.3.6.1.5.5.7.0.35. Is this wrong understanding?

Reviewer 2: @stavamichal my bad, I used https://www.rfc-editor.org/info/rfc3739/#section-3.2.6. I agree that the OID for `qcStatement-2` is the one given by ETSI as `1.3.6.1.5.5.7.0.35.`

Thanks for spotting it! 🙏 

Addendum: actually the correct OID is also referenced in the [rfc](https://datatracker.ietf.org/doc/html/rfc3739.html#appendix-C.1.1.4)

        oid: "2.5.29.31",            // cRLDistributionPoints
        critical: false,
        value: CRLDistributionPoints [
          DistributionPoint {
            distributionPoint: URI("https://crl.example.test/issuing-ca.crl")
          }
        ]
      }
    ]
  },

  signatureAlgorithm: AlgorithmIdentifier {
    oid: "1.2.840.113549.1.1.11",    // must match/align with tbsCertificate.signature
    params: NULL
  },
  signatureValue: "BASE64(SIGN(issuerPrivateKey, DER(tbsCertificate)))"
}
```

##### PuB-EAA Provider Sign/Seal Certificate Content

There are no new or modified parameters specific for PuB-EAA Provider as described in ETSI TS 119 412-6, clause 8.

The following table lists all new or modified extensions that are mandatory or conditional for PuB-EAA Provider as further scoped in ETSI TS 119 412-6, clause 8.2 and 8.3.

| Parameter | Defined in | Presence | Criticality | Format | Description |
| :-------: | :--------: | :------: | :---------: | :----- | :---------- |
| `authorityInfoAccess` | [ETSI TS 119 412-6, clause 8.1] | REQUIRED | NC | *SEQUENCE* | Description is the same as in the General Content above. It is mandatory for PuB-EAA Provider. |

> | `qcStatements` (id-etsi-qcs-QcPSB) | [ETSI TS 119 412-6, clause 8.3] | REQUIRED | NC | *SEQUENCE* | `QCStatement` with the OID `0.4.0.194126.1.3` as defined in [ETSI TS 119 412-6, Annex A]. |

Reviewer 2: I think that there are not two different `qcStatement` extensions, rather a single one where the usual `statementInfo` is substituted by the `QcPSB` element which is a ``SEQUENCE`` containing the following parameters:
- ``countryOfLegislation``: A ``PrintableString (SIZE (2))`` containing the alpha-2 country code of the legislation framework of the public sector body. In the case of European Union law, ``EU`` MUST be used in place of the country code.
- ``authSourceIdentification``: A ``UTF8String`` containing the unique identification of the authentic source.
- ``legislationIdentification``: A ``UTF8String`` containing the identification of the legislation framework.

So the example would be:
```
Extension {
        oid: "1.3.6.1.5.5.7.1.3",   // id-pe-qcStatements container
        critical: false,
        value: QCStatements [
            QCStatement {
                statementId: "0.4.0.194126.1.3", // id-etsi-qcs-QcPSB
                countryOfLegislation: "IT",
                authSourceIdentification: "https://www.anpr.interno.it", // ANPR (Anagrafe Nazionale della Popolazione Residente)
                legislationIdentification: "https://www.normattiva.it/eli/id/2005/05/16/005G0106/sg" // Italian Digital Administration Code (CAD)
            }
        ]
        },
```

Am i missing something?

Author: You are right, I have missed this.

        }
      },

      Extension {
        oid: "2.5.29.14",    // subject key identifier
        critical: false,
        value: SubjectKeyIdentifier [
          keyIdentifier: "SHA-1(SUBJECT_PUBLIC_KEY_VALUE)"
        ]
      },

      Extension {
        oid: "1.3.6.1.5.5.7.1.1",    // authority information access
        critical: false,
        value: AuthorityInfoAccess [
          AccessDescription {
            accessMethod: "1.3.6.1.5.5.7.48.2",            // id-ad-caIssuers
            accessLocation: URI("https://ca.example.test/caIssuers/issuing-ca.cer")
          },

          AccessDescription {
            accessMethod: "1.3.6.1.5.5.7.48.1",            // id-ad-ocsp
            accessLocation: URI("https://ocsp.example.test")
          }
        ]
      },

      Extension {
        oid: "2.5.29.32",            // certificatePolicies
        critical: false,
        value: CertificatePolicies [
          PolicyInformation {
            policyIdentifier: "0.4.0.194112.1.3",          // qcp-legal-qcsd
            policyQualifiers: [
              CPSuri("https://rpca.example.test/cps")
            ]
          }
        ]
      },

      Extension {

>         oid: "1.3.6.1.5.5.7.0.35",   // qcStatements-2 container

Reviewer 2: ```suggestion
        oid: "1.3.6.1.5.5.7.1.3",   // id-pe-qcStatements container
```

        critical: false,
        value: QCStatements [
          QCStatement {

>             statementId: "0.4.0.194126.1.3",   // id-etsi-qcs-QcPSB

Reviewer 2: ```suggestion
            statementId: "0.4.0.194126.1.3",   // id-etsi-qcs-QcPSB
            statementInfo: {
```

>           },

Reviewer 2: ```suggestion
```

>           QCStatement {

Reviewer 2: ```suggestion
```

>             statementId: "0.4.0.1862.1.10",    // esi4-qcStatement-10

Reviewer 2: ```suggestion
```

>             countryOfLegislation: "CZ",

Reviewer 2: ```suggestion
              countryOfLegislation: "CZ",
```

>             authSourceIdentification: "https://authsource.gov.cz/cz/registry/rob",

Reviewer 2: ```suggestion
              authSourceIdentification: "https://authsource.gov.cz/cz/registry/rob",
```

>             legislationIdentification: "https://legislation.gov.cz/eli/cz/sb/2000/365"

Reviewer 2: ```suggestion
              legislationIdentification: "https://legislation.gov.cz/eli/cz/sb/2000/365"
```

>           }

Reviewer 2: ```suggestion
            }
```
